### PR TITLE
Http functionality

### PR DESCRIPTION
**Pull request by lastmjs**

### CODE REVIEW

**Review by bdemann — APPROVED** [review]

File: src/lib/server.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+export const HttpRequest = Record({
+    method: text,
+    url: text,
+    headers: Vec(HeaderField),
+    body: blob,
+    certificate_version: Opt(nat16)
+});
+
+globalThis._azleHttpResponse = HttpResponse;
```
Comment (bdemann) [inline src/lib/server.ts:71]: get rid of these

File: src/lib/server.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+globalThis._azleExportedIc = ic;
+
+let server: NodeServer;
+
+// TODO can the serverInit be async?
+// TODO for example this would be important for sql.js
```
Comment (bdemann) [inline src/lib/server.ts:77]: Do these TODOs

File: src/lib/server.ts [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+                this.responseData = Buffer.concat([
+                    this.responseData,
+                    Buffer.from(data)
+                ]);
+
+                return;
+            }
+
+            if (typeof data === 'string') {
+                console.log('string');
```
Comment (bdemann) [inline src/lib/server.ts:167]: get rid of this log

File: src/lib/server.ts [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+                this.responseData = Buffer.concat([
+                    this.responseData,
+                    Buffer.from(data)
+                ]);
+
+                return;
+            }
+
+            // TODO object is not done, it is not doing anything
+            if (typeof data === 'object') {
```
Comment (bdemann) [inline src/lib/server.ts:177]: do how rust does it and leave a todo

**Review by bdemann — CHANGES_REQUESTED** [review]

File: src/lib/globals.ts [hunk 5]
```diff
@@ -13,6 +14,8 @@ declare global {
     var _azleIcTimers: { [key: string]: string };
     var _azleTimerCallbacks: { [key: string]: () => void };
     var _azleGuardFunctions: { [key: string]: () => any };
+    var _azleExportedIc: typeof ic;
```
[the remark this answers is not in the document]
Comment (lastmjs) [reply src/lib/globals.ts:17]: Resolved

File: package.json [hunk 6]
```diff
@@ -32,11 +32,14 @@
...
         "fs-extra": "10.0.1",
+        "http-message-parser": "^0.0.34",
         "js-sha256": "0.9.0",
+        "net": "^1.0.2",
         "text-encoding": "0.7.0",
         "ts-node": "10.3.1",
         "typescript": "^5.2.2",
-        "uuid": "^9.0.1"
+        "uuid": "^9.0.1",
+        "zlib": "^1.0.5"
```
[the remark this answers is not in the document]
Comment (lastmjs) [reply package.json:42]: Resolved

File: src/compiler/compile_typescript_code.ts [hunk 7]
```diff
@@ -83,11 +83,13 @@ export function bundleFromString(ts: TypeScript): JavaScript {
             querystring: `${GLOBAL_AZLE_WASMEDGE_QUICKJS_DIR}/modules/querystring.js`,
             whatwg_url: `${GLOBAL_AZLE_WASMEDGE_QUICKJS_DIR}/modules/whatwg_url.js`,
             encoding: `${GLOBAL_AZLE_WASMEDGE_QUICKJS_DIR}/modules/encoding.js`,
+            http: `${GLOBAL_AZLE_WASMEDGE_QUICKJS_DIR}/modules/http.js`,
             crypto: 'crypto-browserify',
+            zlib: 'crypto-browserify', // TODO wrong of course
```
[the remark this answers is not in the document]
Comment (lastmjs) [reply src/compiler/compile_typescript_code.ts:88]: Tried and it just broke, so leaving it

File: src/lib/server.ts (hunk 2 above)
Comment (lastmjs) [reply src/lib/server.ts:77]: I did this in the next PR

**Review by bdemann — APPROVED** [review]

**Pull request merged**